rapid7/metasploit-framework
# Northstar C2 Stored XSS to Agent RCE (CVE-2024-28741)

This PR adds a new exploit against Northstar C2, exploiting CVE-2024-28741. We're able to pretend to be an infected computer (agent) and send an XSS payload. When a logged in user views the logs, we take their cookie, then exploit all the agents to deploy our own shellcode.

## Verification

1. Install the application, and connect an agent
1. Start msfconsole
1. Do: `use exploit/windows/http/northstar_c2_xss_to_agent_rce`
1. Do: `set rhosts [ip]`
1. Do: `set srvhost [srvhost]`
1. Do: `set fetch_srvport [fetch_srvport]`
1. Do: `set fetch_srvhost [fetch_srvhost]`
1. Do: `run`
1. Do: visit the NorthStarC2 site with a logged in user, and browse to the Server Logs page.
1. You should get a shell on each agent.
